src/main/java/net/rubygrapefruit/platform/internal/jni/AbstractFileEventFunctions.java

```diff
@@ -108,14 +120,26 @@ public boolean stopWatching(Collection<File> paths) {
         }
 
         @Override
-        public void close() {
-            if (server == null) {
-                throw new NativeException("Closed already");
-            }
+        public void close() throws InterruptedIOException {
```

Comment:
I chose to throw `InterruptedIOException` because `close()` by contract can only throw `IOException`. We can use `InterrputedException` once we implement `shutdown()` / `awaitTermination()` in #183.

Good observation about the `InterruptedException` not signaling the thread being left alive. We should indeed handle that.

Reply:
If you don't throw the `InterruptedException`, then you need to clear the interrupted flag.